graphql-spring-boot-autoconfigure/src/main/java/graphql/kickstart/autoconfigure/web/servlet/AsyncServletProperties.java

```diff
@@ -1,14 +1,25 @@
 package graphql.kickstart.autoconfigure.web.servlet;
 
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
 import lombok.Data;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.convert.DurationUnit;
+import org.springframework.lang.NonNull;
+import org.springframework.validation.annotation.Validated;
 
 @Data
 @ConfigurationProperties(prefix = "graphql.servlet.async")
+@Validated
 public class AsyncServletProperties {
 
   private boolean enabled = true;
-  private long timeout = 30000;
+  /**
+   * Asynchronous execution timeout. If a duration suffix is not specified, millisecond will be used.
+   */
+  @NonNull
```

Comment:
actually `javax.validation.constraints.NotNull` is needed to make validation work.

But it seems that even if I try to override the default value with `null` from config, the null value seems to be ignored and the default value will still be used and this validation may not trigger, even with the correct annotation...

So it may not be necessary after all.

Reply:
Finally, I decided to use the lombok.NotNull annotation only here

Reply:
I would prefer to use validation constraints for validation (Lombok `@NonNull` would still result in NPE). But let's skip the validation part for now. Property validation should be consistently handled across the project, and that would be a topic for a separate task / PR.